web_tree_date_search/static/src/js/web_tree_date_search.js

```diff
@@ -0,0 +1,159 @@
+                    for (col in this.columns){
+                        if (this.columns[col].name == date_field){
+                            this.date_field_data[date_field] = []
+                            this.date_field_data[date_field][0] = this.columns[col].string;
+                            this.date_field_data[date_field][1] = this.columns[col].type;
+                            break;
+                        }
+                    }
+                }
+                if (this.dates_filter && this.dates_filter.length > 0){
+                    this.$el.parent().prepend(QWeb.render('web_tree_date_search_toolbar', {widget: this}));
+                    var ui_toolbar_loc = $('.ui-toolbar:last');
+                    ui_toolbar_loc.show();
+                    for (i in this.dates_filter){
+                        var date_field = this.dates_filter[i];
+                        var date_string = this.date_field_data[date_field][0]
+                        var date_type = this.date_field_data[date_field][1]
+
+                        self.columns
+                        var date_div =  QWeb.render('web_tree_date_search_field', {'field_name': date_string});
+                        var toolbar_height = ui_toolbar_loc.height();
+
+                        ui_toolbar_loc.append(date_div);
+                        $('div.oe_form_dropdown_section:last span:eq(0)').addClass('oe_date_filter_from_' + date_field);
+                        $('div.oe_form_dropdown_section:last span:eq(1)').addClass('oe_date_filter_to_' + date_field);
+
+                        this.value = new (instance.web.search.custom_filters.get_object(date_type))
+                            (this, {"selectable":true,
+                                "name":"oe_date_filter_from_" + date_field, 
+                                "type": date_type,
+                                "string":date_string});
+                        var value_loc = $('.oe_date_filter_from_' + date_field + ':last').show().empty();
+                        this.value.appendTo(value_loc);
+
+                        this.value = new (instance.web.search.custom_filters.get_object(date_type))
+                            (this, {"selectable":true,
+                                "name":"oe_date_filter_to_" + date_field, 
+                                "type": date_type,
+                                "string":date_string});
+                        var value_loc = $('.oe_date_filter_to_' + date_field + ':last').show().empty();
+                        this.value.appendTo(value_loc);
+
+                        var oe_date_filter_from = $('.oe_date_filter_from_' + date_field + ':last .oe_datepicker_master');
+                        var oe_date_filter_to = $('.oe_date_filter_to_' + date_field + ':last .oe_datepicker_master');
+                        if (date_type == 'date'){
+                            oe_date_filter_from.css("width", "105px");
```

Comment:
It would be better to create a specific class for this instead of static value and then use the class in code so that if anyone wants to change the value, he/she can change it by extending the class.